topiary/tests/samples/expected/ocaml.ml

```diff
@@ -13,6 +13,24 @@
 (*                                                                        *)
 (**************************************************************************)
 
+(* Issue 529 *)
+
+let id1 = if true then Fun.id else function x -> x
+
+let id2 = if true then Fun.id else function x -> x | y -> y
+
+let id3 =
+  if true then
+    Fun.id
+  else
+    function x -> x
+
+let id4 = function x -> x
+
+let id5 = if true then Fun.id else function | x -> x
```

Comment:
We should be able to drop the bar here. See issue for discussion.

Reply:
The bar should be dropped, now that #537 is merged. I think we can also keep the spacing after `fun` and `function`, for additional guarantees.